experimenter/experimenter/changelog/views.py

```diff
@@ -1 +1,27 @@
-# Create your views here.
+from django.shortcuts import get_object_or_404
+from django.views.generic import DetailView
+
+from experimenter.experiments.models import NimbusExperiment
+
+
+class NimbusChangeLogsView(DetailView):
+    model = NimbusExperiment
+    template_name = "changelog/overview.html"
+    context_object_name = "experiment"
+
+    def get_object(self, queryset=None):
+        slug = self.kwargs.get("slug")
+        experiment = get_object_or_404(NimbusExperiment, slug=slug)
+        return experiment
+
+    def get_context_data(self, **kwargs):
+        slug = self.kwargs.get("slug")
+        context = super().get_context_data(**kwargs)
+        experiment = self.get_object()
```

Comment:
This is unnecessary. The view will populate the context with `[context_object_name] = get_object()` [src](https://github.com/django/django/blob/ddb6506618ea52c6b20e97eefad03ed847a1e3de/django/views/generic/detail.py#L99)

Reply:
So should I remove the `get_object` method from the view and have just the `get_context_data` method like this 
```
def get_context_data(self, **kwargs):
    context = super().get_context_data(**kwargs)
    context['changelogs'] = list(self.object.changes.all())
    return context
```
or just remove all the methods altogether and access the changelogs directly like 
```
for changelog in nimbusexperiment.changes.all
```
But I guess the latter will have issues as I'm directly iterating a queryset right ? // @brennie

Reply:
Pre-fetching changelogs in get_context_data is fine